### PR TITLE
Pv2 1911 change of is levy flag0to1

### PR DESCRIPTION
**Pull request by keenesolutions**

### CODE REVIEW

File: src/SFA.DAS.Payments.FundingSource.Application/Services/ManageLevyAccountBalanceService.cs [hunk 1]
```diff
@@ -115,19 +113,31 @@ private List<LevyAccountModel> MapToLevyAccountModel(PagedApiResponseViewModel<A
...
+                .Where(x => !storedEmployers.Contains((x.AccountId, x.IsLevyPayer))).ToList();
 
             var endpointInstance = await endpointInstanceFactory.GetEndpointInstance().ConfigureAwait(false);
-            var publishMessageTasks = notLevyPayingEmployerIds.Select(accountId => endpointInstance.Publish(new FoundNotLevyPayerEmployerAccount { AccountId = accountId }));
-            await Task.WhenAll(publishMessageTasks).ConfigureAwait(false);
+            List<Task> publishEvents = new List<Task>();
 
-            logger.LogInfo($"Successfully Published FoundNotLevyPayerEmployerAccount event for  Account Ids: {string.Join(",", notLevyPayingEmployerIds)}");
-        }
+            publishEvents.AddRange(accountsWithChangedLevyFlags.Where(x=> !x.IsLevyPayer).Select(employer => endpointInstance.Publish(new FoundNotLevyPayerEmployerAccount { AccountId = employer.AccountId })));
```
Comment (hardik-desai-kcl) [inline src/SFA.DAS.Payments.FundingSource.Application/Services/ManageLevyAccountBalanceService.cs:131]: I haven't tested this but could be rewritten as
 var publishEvents =
                accountsWithChangedLevyFlags.Select(employer => 
                                                        endpointInstance.Publish( !employer.IsLevyPayer 
                                                                                      ? (object) new FoundNotLevyPayerEmployerAccount { AccountId = employer.AccountId }
                                                                                      : (object) new FoundLevyPayerEmployerAccount { AccountId = employer.AccountId } ));

File: src/SFA.DAS.Payments.DataLocks.DataLockProxyService/Handlers/FoundLevyPayerEmployerAccountEventHandler.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        {
+            this.proxyFactory = proxyFactory;
+            this.paymentLogger = paymentLogger;
+            this.scope = scope;
+        }
+
+        public async Task Handle(FoundLevyPayerEmployerAccount message, IMessageHandlerContext context)
+        {
+            paymentLogger.LogDebug($"Handling Found LevyPayer Employer Account Id: {message.AccountId}");
+            var processor = scope.Resolve<IApprenticeshipProcessor>();
```
[the remark this answers is not in the document]
Comment (keenesolutions) [reply src/SFA.DAS.Payments.DataLocks.DataLockProxyService/Handlers/FoundLevyPayerEmployerAccountEventHandler.cs:30]: I've done this here and also for the counterpart handler that it was based on.

File: src/SFA.DAS.Payments.DataLocks.Application/Services/ApprenticeshipProcessor.cs [hunk 3]
```diff
@@ -243,30 +243,30 @@ public async Task ProcessPaymentOrderChange(PaymentOrderChangedEvent paymentOrde
             }
         }
 
-        public async Task ProcessApprenticeshipForNonLevyPayerEmployer(long accountId)
+        public async Task ProcessNonLevyPayerFlagForEmployer(long accountId, bool isLevyPayer)
```
[the remark this answers is not in the document]
Comment (keenesolutions) [reply src/SFA.DAS.Payments.DataLocks.Application/Services/ApprenticeshipProcessor.cs:246]: I've renamed this to ProcessIsLevyPayerFlagForEmployer which I think is even better

File: src/SFA.DAS.Payments.FundingSource.Application/Repositories/LevyFundingSourceRepository.cs [hunk 4]
```diff
@@ -52,6 +52,13 @@ public async Task<List<long>> GetEmployerAccounts(CancellationToken cancellation
             return accounts.Distinct().ToList();
         }
 
+        public async Task<List<(long, bool)>> GetCurrentEmployerStatus(List<long> employerIds,CancellationToken cancellationToken = default(CancellationToken))
```
[the remark this answers is not in the document]
Comment (keenesolutions) [reply src/SFA.DAS.Payments.FundingSource.Application/Repositories/LevyFundingSourceRepository.cs:55]: I'm not sure about this change. I think the tuple contains is probably as easy to read as the linq query syntax join statement. Also, this adds filtering logic in the repository which no other repositiory methods do.

File: src/SFA.DAS.Payments.FundingSource.Application/Services/ManageLevyAccountBalanceService.cs (hunk 1 above)
Comment (keenesolutions) [reply src/SFA.DAS.Payments.FundingSource.Application/Services/ManageLevyAccountBalanceService.cs:131]: 👍 I made this change as it's much clearer. I've only cast one of them to object to keep resharper happy.

**Pull request merged**